src/static/helpers/customApi.ts

```diff
@@ -0,0 +1,150 @@
+ * @param args.options.body? - Body that is used for the request
+ * @param args.clientConfig - Client Configuration object used by the SDK with properties that can affect the fetch call
+ * @param args.clientConfig.parameters - Path parameters used for custom API endpoints. The required properties are: apiName, endpointPath, organizationId, and shortCode. An error will be thrown if these are not provided.
+ * @param args.clientConfig.headers? - Additional headers that are added to the request. Authorization header should be in this argument or in the options?.headers. options?.headers will override any duplicate properties. If "Content-Type" is not provided in either header, it will be defaulted to "application/json".
+ * @param args.clientConfig.baseUri? - baseUri used for the request, where the path parameters are wrapped in curly braces. Default value is 'https://{shortCode}.api.commercecloud.salesforce.com/custom/{apiName}/{apiVersion}'
+ * @param args.clientConfig.fetchOptions? - fetchOptions that are passed onto the fetch request
+ * @param args.clientConfig.throwOnBadResponse? - flag that when set true will throw a response error if the fetch request fails (returns with a status code outside the range of 200-299 or 304 redirect)
+ * @param args.clientConfig.proxy? - Routes API calls through a proxy when set
+ * @param args.rawResponse? - Flag to return the raw response from the fetch call. True for raw response object, false for the data from the response
+ * @returns Raw response or data from response based on rawResponse argument from fetch call
+ */
+export const callCustomEndpoint = async (args: {
+  options: {
+    method?: string;
+    parameters?: {
+      [key: string]: string | number | boolean | string[] | number[];
+    };
+    customApiPathParameters?: {
+      apiName?: string;
+      apiVersion?: string;
+      endpointPath?: string;
+      organizationId?: string;
+      shortCode?: string;
+    };
+    headers?: {
+      authorization?: string;
+    } & {[key: string]: string};
+    body?: BodyInit | globalThis.BodyInit | unknown;
+  };
+  clientConfig: ClientConfigInit<CustomParams>;
+  rawResponse?: boolean;
+}): Promise<Response | unknown> => {
+  const {options, clientConfig, rawResponse} = args;
+
+  const requiredArgs = [
+    'apiName',
+    'endpointPath',
+    'organizationId',
+    'shortCode',
+  ];
+
+  const pathParams: Record<string, unknown> = {
+    ...clientConfig.parameters,
+    ...options?.customApiPathParameters,
+  };
+
+  requiredArgs.forEach(arg => {
+    if (!pathParams[arg]) {
+      throw new Error(
+        `Missing required property needed in options.customApiPathParameters or clientConfig.parameters: ${arg}`
+      );
+    }
+  });
+
+  if (!pathParams.apiVersion) {
+    pathParams.apiVersion = 'v1';
+  }
+
+  const defaultBaseUri =
```

Comment:
We had the conversation about the baseUri before. It doesn't look like we are going to be do anything about handling non-prod uri's. Is that the case, and have you shared that decision with anyone else or documented it.